---
layout: post
title: 利用Long Ranger分析10X测序数据(二)
category: 技术
tags: [bioinfo, linux, 10x, long ranger]
notebook: bioinfo
---

{% cq %}Perseverance is permanent enjoy.

--Black
{% endcq %}

## 1. Long Ranger调用bcl2fastq

如果需要转换bcl产生FASTQ文件，则需要使用[Illumina's bcl2fastq 2.17](http://support.illumina.com/downloads/bcl2fastq_conversion_software.html)。安装Long Ranger与bcl2fastq，并加入到`$PATH`变量中即可调用 `longranger mkfastq`转化bcl文ä»¶。

### 1.1 bcl2fastq转换

命令：

```
bcl2fastq -R /BHW5F3CCXX/ --sample-sheet ./SampleSheet.csv -o ./1lane.fq/
```

### 1.2 longranger 转换

```
$ longranger mkfastq --run=/path/to/tiny_bcl \
                     --samplesheet=tiny-bcl-samplesheet.csv

longranger mkfastq
Copyright (c) 2016 10x Genomics, Inc.  All rights reserved.
-----------------------------------------------------------------------------
Martian Runtime - 2.1.1

Running preflight checks (please wait)...
```

转化前文件结构示例：

```
./
├── Data
│   └── Intensities
│       ├── BaseCalls
│       │   └── L004
│       │       ├── C100.1
│       │       │   ├── s_4_1102.bcl.gz
│       │       │   └── s_4_2224.bcl.gz
│       │       ├── C10.1
│       │       │   ├── s_4_2223.bcl.gz
│       │       │   └── s_4_2224.bcl.gz
│       │       ├── s_4_1101.filter
│       │Â       └── s_4_2224.filter
│       └── s.locs
├── InterOp
│   ├── StaticRunMetricsOut.bin
│   └── TileMetricsOut.bin
├── RunInfo.xml
└── SampleSheet.csv
```

### 1.3 参考文献

 - [http://support.10xgenomics.com/genome-exome/software/pipelines/latest/using/mkfastq](http://support.10xgenomics.com/genome-exome/software/pipelines/latest/using/mkfastq)

## 2. WGS全基因组数据分析

### 2.1 数据demultiplexe和格式转换

调用`longranger mkfastq` or `bcl2fastq` 转化FASTQ文件。

### 2.2 在WGS模式下运行longranger

```
$ longranger wgs --id=sample345 \
                 --reference=/opt/refdata-hg19-2.1.0 \
                 --fastqs=/home/jdoe/runs/HAWT7ADXX/outs/fastq_path
```

| Parameter      |    Description |
| :-------- | :--------|
| --id  | A unique run ID string: e.g. sample345  |
| --fastqs  | Path of the FASTQ folder generated by longranger mkfastq
e.g. /home/jdoe/runs/HAWT7ADXX/outs/fastq_path  |
| -sample  | (optional) Sample name as specified in the sample sheet supplied to mkfastq.  |
| --reference  | Path to a 10x compatible reference, e.g. /opt/refdata-hg19-2.1.0.  |
| --vcmode  | (optional) freebayes, gatk:/path/to/GenomeAnalysisTK.jar, or disable  |
| --precalled  | (optional) Path to a "pre-called" VCF file. Variants in this file will be phased.  |
| --sex  |  (optional) Sex of the sample: male or female. Sex will be detected based on coverage if not supplied.  |
| --somatic  | (optional) Supply this flag for somatic samples. This will increase the sensitivity of the large-scale SV caller. Note: this option currently does not affect small-scale variant calling. The small scale variant caller is not currently optimized for somatic variants  |

其他参数：`--localcores=16` 默认情况下将会使用所有的CPU，该参数能够限制使用的cpu核数。`--localmem` 将会限制程序使用的内存大小（**GB**）。

### 2.3 Variant Calling

默认情况下，longranger将会调用freebayes进行 call SNP 和 indels， phase variants, adds structural variant calls。也可以通过*--vcmode*指定GATK进行SNP检测。

也可以利用`--pre-called`参数，引入已经处理完的**VCF**文件。

### 2.4 Output Files

Pipleline运行结束后，产生的文件结æ如下：

```
2016-05-02 15:46:41 [runtime] (run:local)       ID.sample345.PHASER_SVCALLER_CS.PHASER_SVCALLER.LOUPE_PREPROCESS.fork0.join
2016-05-02 15:46:44 [runtime] (join_complete)   ID.sample345.PHASER_SVCALLER_CS.PHASER_SVCALLER.LOUPE_PREPROCESS
2016-05-02 15:46:55 [runtime] VDR killed 4738 files, 223GB.

Outputs:
- Run summary:               /home/jdoe/runs/sample345/outs/summary.csv
- BAM barcoded:              /home/jdoe/runs/sample345/outs/phased_possorted_bam.bam
- BAM index:                 /home/jdoe/runs/sample345/outs/phased_possorted_bam.bam.bai
- VCF phased:                /home/jdoe/runs/sample345/outs/phased_variants.vcf.gz
- VCF index:                 /home/jdoe/runs/sample345/outs/phased_variants.vcf.gz.tbi
- Large-scale SV calls:      /home/jdoe/runs/sample345/outs/large_sv_calls.bedpe
- Large-scale SV candidates: /home/jdoe/runs/sample345/outs/large_sv_candidates.bedpe
- Large-scale SVs:           /home/jdoe/runs/sample345/outs/large_svs.vcf.gz
- Large-scale SVs index:     /home/jdoe/runs/sample345/outs/large_svs.vcf.gz.tbi
- Mid-scale deletions:       /home/jdoe/runs/sample345/outs/dels.vcf.gz
- Mid-scale deletions index: /home/jdoe/runs/sample345/outs/dels.vcf.gz.tbi
- Loupe file:                /home/jdoe/runs/sample345/outs/loupe.loupe

Pipestance completed successfully!
```

| File Name  | Description |
| :-------- | :--------|
|summary.csv  | Run summary metrics in CSV format |
|phased_possorted_bam.bam  |  Aligned reads annotated with barcode information ||
|phased_possorted_bam.bam.bai  |  Index for phased_possorted_bam.bam |
|phased_variants.vcf.gz  |  VCF annotated with barcode and phasing information ||
|phased_variants.vcf.gz.tbi  |  Index for phased_variants.vcf.gz |
|large_sv_calls.bedpe  |  Confidently called large-scale structural variants (|≥30Kbp or inter-chromosomal) in BEDPE format |
|large_sv_candidates.bedpe  | Large-scale structural variant calls and low |confidence candidates in BEDPE format |
|large_svs.vcf.gz  |  Large-scale structural variant calls and candidates in |VCF format |
|large_svs.vcf.gz.tbi  |  Index for large_svs.vcf.gz |
|dels.vcf.gz | Mid-scale deletion calls (50bp-30Kbp) |
|dels.vcf.gz.tbi |  Index for dels.vcf.gz |
|loupe.loupe  | File that can be opened in the Loupe genome browser |

运行结束后，可以使用[Loupe genome browser](http://support.10xgenomics.com/genome-exome/software/visualization/latest/getting-started)软件查看loupe文件。






